### ISSUE TITLE
Stack Counter

### ISSUE THREAD

**Issue by irlanrobson** [RandyGaul/cute_framework#83]
Shouldn't the stack counter in this line be less or equal capacity?

https://github.com/RandyGaul/cute_framework/blob/master/src/cute_aabb_tree.cpp#L784

E.g.:

`CUTE_ASSERT(sp <= AABB_TREE_STACK_QUERY_CAPACITY);
`

**Comment by RandyGaul** [comment]
Yes definitely. I should actually add in a task to make use of the `apush` and `astatic`. I'll just modify this topic for tracking the issue.

**Comment by RandyGaul** [comment]
https://github.com/RandyGaul/cute_framework/commit/734081a3a6d5e9954519ab601f26719a7160c8b0
